rfcs/2021-07-20-8288-csv-enrichment.md

```diff
@@ -0,0 +1,202 @@
+# RFC 8288 - 2021-07-20 - CSV Enrichment
+
+Vector needs to allow users to enrich the events flowing through the topology
+using a CSV file. This RFC proposes adding a VRL function that can perform 
+this lookup.
+
+## Scope
+
+The RFC will cover a simple VRL function that will lookup a single row from a
+CSV file. 
+
+The lookup will be a simple sequential scan. The first version of this
+function will not handle any indexing of this data. Above a certain size
+indexing will provide substantial performance benefits, so it is something that
+we should definitely plan for implementing at a future stage. The RFC will 
+briefly discuss this, but will not go into any specific indexing strategies.
+
+## Motivation
+
+Users would like to enrich events flowing through Vector with extra data 
+provided by a CSV file. 
+
+## Doc-level Proposal
+
+### Resources
+
+To represent the CSV file we have a new top level configuration option.
+
+```
+[resources.csv_file]
+  type = "csv"
+  path = "\path_to_csv"
+  key = "{KEY}"
+  delimeter = ","
```

Comment:
I'm not a fan of this. It feels premature to define an abstract concept like "resources" just for CSV data. A simpler approach would be to supply these details as arguments to the VRL function. We can still load the file as a "resource" during compilation time, but the user isn't required to jump through those hoops.